src/client/datascience/data-viewing/data-wrangler/dataWrangler.ts

```diff
@@ -0,0 +1,685 @@
+                code = historyItem.code;
+                newVariableName = historyItem.variableName;
+                break;
+
+            case DataWranglerCommands.GetHistoryItem:
+                this.getHistoryItem(payload.args.index).ignoreErrors();
+                break;
+
+            case DataWranglerCommands.Describe:
+                void this.getColumnStats(payload.args.columnName);
+                break;
+        }
+
+        const dataCleaningMode = this.configService.getSettings().dataCleaningMode;
+        if (dataCleaningMode === OpenDataWranglerSetting.STANDALONE) {
+            if (code && notebook !== undefined) {
+                void notebook?.execute(code, '', 0, uuid()).then(async () => {
+                    if (this.existingDisposable) {
+                        this.existingDisposable.dispose();
+                    }
+                    await this.updateWithNewVariable(newVariableName);
+                });
+            }
+        } else if (dataCleaningMode === OpenDataWranglerSetting.WITH_JUPYTER_NOTEBOOK) {
+            if (code && matchingNotebookEditor !== undefined) {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                let cells = (matchingNotebookEditor as any).document.getCells();
+                let lastCell = cells[cells.length - 1] as NotebookCell;
+                await addNewCellAfter(lastCell, '');
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                cells = (matchingNotebookEditor as any).document.getCells();
+                lastCell = cells[cells.length - 1] as NotebookCell;
+                await updateCellCode(lastCell, code);
+                if (this.existingDisposable) {
+                    this.existingDisposable.dispose();
+                }
+                this.existingDisposable = vscNotebook.onDidChangeNotebookCellExecutionState(
+                    async (e: NotebookCellExecutionStateChangeEvent) => {
+                        if (e.state === NotebookCellExecutionState.Idle && refreshRequired) {
+                            await this.updateWithNewVariable(newVariableName);
+                        }
+                    }
+                );
+
+                await this.commandManager.executeCommand(
+                    'notebook.cell.execute',
+                    { start: lastCell.index, end: lastCell.notebook.cellCount },
+                    lastCell.notebook.uri
+                );
+            }
+        }
+    }
+
+    private renameColumn(currentVariableName: string, req: IRenameColumnsRequest): IHistoryItem {
+        this.variableCounter += 1;
+        const newVariableName = `df${this.variableCounter}`;
+        const code = `${newVariableName} = ${currentVariableName}.rename(columns={ "${req.oldColumnName}": "${req.newColumnName}" })\n`;
+        const historyItem = {
+            transformation: `Renamed column "${req.oldColumnName}" to "${req.newColumnName}"`,
```

Comment:
I believe these 'actions' need to be localized.

Reply:
Added to localize.ts and package.nls.json